### DISCUSSION TITLE
How to set up referential action for one-to-one relations

### DISCUSSION THREAD

**Discussion by codewitch** [prisma/prisma#15287] [Q&A] [answered]
I have two models
```
model Foo {
   id       String      @id @default(auto()) @map("_id") @db.ObjectId
   title    String
   bar     Bar?
}

model Bar {
   id        String      @id @default(auto()) @map("_id") @db.ObjectId
   fooId  String       @db.ObjectId
   foo     Foo           @relation(fields: [fooId], references: [id], onDelete: Cascade)
}
```

But when I delete an instance of Bar, the related Foo still remains in the database. Does this referential rule only work in one direction? I.e. only when you delete Foo, the related Bar will delete? How can I make it so deleting a Bar instance cascades to Foo?

I tried adding a `@relation(onDelete: Cascade` to the Foo schema, but that triggered a prisma schema error.

**Comment by janpio** [comment, marked as answer]
Referential actions always only work "in one direction". If you want the other direction, you will need to swap the relation. If you want both directions, you will need to have relations in both directions.

**Reply by codewitch** [reply]
thank you! That is good to know. I will add logic to explicitly delete the related collection for this since it will be cumbersome to migrate existing data. But this will inform us in making better schema designs for future models!
